cupy/_statistics/correlation.py

```diff
@@ -131,13 +145,62 @@ def cov(a, y=None, rowvar=True, bias=False, ddof=None):
     if ddof is None:
         ddof = 0 if bias else 1
 
-    fact = X.shape[1] - ddof
+    w = None
+    if fweights is not None:
+        fweights = cupy.asarray(fweights, dtype=float)
+        if not cupy.all(fweights == cupy.around(fweights)):
```

Comment:
This causes device synchronization. It is better to check just the type info, `fweights.dtype.char in 'bBhHiIlLqQ'`

Reply:
Which part exactly causes device sync?